File: models/Player.rb
class Player
    attr_accessor :id, :firstName, :lastName, :primaryNumber
    attr_accessor :birthYear, :birthMonth, :birthDay, :birthCity, :birthProvince, :birthCountry
    attr_accessor :height, :weight, :active, :shoot, :rookie, :teamId, :positionCode

    def fullName
        return "#{firstName} #{lastName}"
    end

    def birthDate
        return "#{birthYear}-#{birthMonth}-#{birthDay}"
    end

    def birthLocation
        location = birthCity

        if birthProvince |= nil
            location += ", #{birthProvince}"
        end

        location += ", #{birthCountry}"
    end

    def self.fromJson(item)
        # p item
        newPlayer = Player.new

        newPlayer.id = item["id"]
        newPlayer.firstName = item["firstName"]
        newPlayer.lastName = item["lastName"]
        newPlayer.primaryNumber = item["primaryNumber"]

        birthDate = item["birthDate"]
        birthDateParts = birthDate.split("-")
        newPlayer.birthYear = birthDateParts[0]
        newPlayer.birthMonth = birthDateParts[1]
        newPlayer.birthDay = birthDateParts[2]
        newPlayer.birthCity = item["birthCity"]
        newPlayer.birthProvince = item["birthStateProvince"]
        newPlayer.birthCountry = item["birthCountry"]

        newPlayer.height = item["height"]
        newPlayer.weight = item["weight"]
        newPlayer.active = item["active"]
        newPlayer.shoot = item["shootsCatches"]
        newPlayer.rookie = item["rookie"]
        newPlayer.teamId = item["currentTeam"]["id"]
        newPlayer.positionCode = item["primaryPosition"]["code"]

        return newPlayer
    end

    def self.fromPoolRow(row)
        player = Player.new

        player.id = row["id"]
        player.firstName = row["firstName"]
        player.lastName = row["lastName"]
        player.positionCode = row["positionCode"]

        return player
    end

    def self.fromRow(row)
        player = Player.new

        player.id = row["id"]

        player.firstName = row["firstName"]
        player.lastName = row["lastName"]
        player.primaryNumber = row["primaryNumber"]

        player.birthYear = row["birthYear"]
        player.birthMonth = row["birthMonth"]
        player.birthDay = row["birthDay"]

        player.birthCity = row["birthCity"]
        player.birthProvince = row["birthStateProvince"]
        player.birthCountry = row["birthCountry"]

        player.positionCode = row["positionCode"]

        return player
    end
end

File: src/main/kotlin/ca/sebleclerc/hockeydata/models/Player.kt
package ca.sebleclerc.hockeydata.models

import java.sql.ResultSet

data class Player(
  val id: Int,
  val firstName: String,
  val lastName: String,
  val primaryNumber: Int,
  val birthDate: BirthDate,
  val height: Int,
  val weight: Int,
  val shoot: String,
  val rookie: Boolean,
  val teamId: Int,
  val positionCode: String,
  val headshotUrl: String,
  val heroImageUrl: String
) {
  companion object
}

// Row
fun Player.Companion.fromRow(rs: ResultSet): Player {
  return Player(
    rs.getInt("id"),
    rs.getString("firstName"),
    rs.getString("lastName"),
    rs.getInt("primaryNumber"),
    BirthDate.fromRow(rs),
    rs.getInt("height"),
    rs.getInt("weight"),
    rs.getString("shoot"),
    rs.getBoolean("rookie"),
    rs.getInt("teamId"),
    rs.getString("positionCode"),
    rs.getString("headshotUrl"),
    rs.getString("heroImageUrl"),
  )
}